### ISSUE TITLE
pedestrian avoids sidewalk and takes primary

### ISSUE THREAD

**Issue by tatobox** [valhalla/valhalla#4657]
Hi everyone,
When I create a walking route the track goes on the road and does not follow the sidewalk.
In URL just a short example but it happens often. Is this a problem that needs to be corrected in iD Editor or a routing problem in Valhalla?
With OSMAnd on iOS the routing is correct
[https://valhalla.openstreetmap.de/directions?profile=pedestrian&wps=11.325772404670715%2C44.4969071020134%2C11.327086687088014%2C44.497898125113466](url)
It should look like the blue line in the screenshot.
<table style="width:100%">
  <tr>
    <td><img src="https://github.com/gis-ops/valhalla-app/assets/153958793/c6160e35-a3cb-4da6-8792-dfed2d082486)"  width="300" height="auto"/> </td>
    <td><img src="https://github.com/gis-ops/valhalla-app/assets/153958793/def055da-391c-4f0d-b0db-4c4371036d8c)"  width="300" height="auto"/> </td>
  </tr>
</table>

**Comment by nilsnolde** [comment]
> With OSMAnd on iOS the routing is correct

then it's likely ok connected (unless osmand (they're still using their own routing engine?!) does some connectivity tricks).

next thing to look at would be how well our preference for sidewalks works. your suggested path might be a bit longer?

**Comment by nilsnolde** [comment]
wtf github, can't click on the link, it mangles the URL somehow to `https://github.com/valhalla/valhalla/issues/url` :roll_eyes:

**Comment by nilsnolde** [comment]
maybe we have an issue with "unmarked" crossings? this here routes just fine: 
https://valhalla.openstreetmap.de/directions?profile=pedestrian&wps=11.326072812080383%2C44.496884143749696%2C11.327290534973145%2C44.49738157077296

but as soon as you have to traverse this crossing, it takes the primary (!): https://www.openstreetmap.org/way/1169674507

**Comment by nilsnolde** [comment]
yeah, real quick look and seems we don't treat those crossing as a sidewalk anymore which we probably should. because then it'd probably be preferred.

ref lua:
https://github.com/valhalla/valhalla/blob/962719a352d06198dfbfae2c653c00f6aa99e060/lua/graph.lua#L1506-L1511

pbfparser:
https://github.com/valhalla/valhalla/blob/962719a352d06198dfbfae2c653c00f6aa99e060/src/mjolnir/pbfgraphparser.cc#L395-L397

that `Use::kPedestrianCrossing` is only used in narration, but never in costing, only `kFootway` and `kSidewalk` are:
https://github.com/valhalla/valhalla/blob/962719a352d06198dfbfae2c653c00f6aa99e060/src/sif/pedestriancost.cc#L719-L721

any opinions on this @gknisely @dnesbitt61 ?

**Comment by tatobox** [comment]
> maybe we have an issue with "unmarked" crossings? this here routes just fine: https://valhalla.openstreetmap.de/directions?profile=pedestrian&wps=11.326072812080383%2C44.496884143749696%2C11.327290534973145%2C44.49738157077296
> 
> but as soon as you have to traverse this crossing, it takes the primary (!): https://www.openstreetmap.org/way/1169674507

Probably yes.
Just to give another example from a little further away in "Via del Carso" avoid the crossing and go on the road

https://valhalla.openstreetmap.de/directions?profile=pedestrian&wps=11.322559118270876%2C44.49704102503823%2C11.327118873596191%2C44.49790960404683

**Comment by tatobox** [comment]
However, if I look better on iD, the "Via del Carso" crossing is a "marked" crossing but it still goes on the road

**Comment by nilsnolde** [comment]
Yeah I could imagine that there’s quite a few connectivity issues with sidewalks in OSM. It’s also hard business on busy intersections. 

At some point someone will look at this, it’s pretty annoying when it doesn’t use the nice sidewalks that exist.

**Comment by tatobox** [comment]
OSMAnd works perfectly.
<table style="width:100%">
  <tr>
    <td><img src="https://github.com/valhalla/valhalla/assets/153958793/2ccef80c-f5c9-43e0-86be-841b40ab5e87)"  width="600" height="auto"/> </td>
  </tr>
<tr>
    <td><img src="https://github.com/valhalla/valhalla/assets/153958793/4d86fc4a-1549-49b3-b12a-9ec5357729e9)"  width="600" height="auto"/> </td>
  </tr>
</table>
